### DISCUSSION TITLE
How to enable "time" feature in i2c.rs?

### DISCUSSION THREAD

**Discussion by albertskog** [ch32-rs/ch32-hal#65] [Q&A]
Hello, this feels like a silly question but how do I set up my project to enable the `time` feature in i2c.rs? From what I understand, I need to have an embassy-time-driver enabled and that in turn should provide it? I assumed this is what the feature `"time-driver-tim2"` from the examples should do, but it is not working in my case. 

Here are my dependencies:
```toml
[dependencies]
panic-halt = "1.0.0"
ch32-hal = { git = "https://github.com/ch32-rs/ch32-hal", rev = "0468b48", features = [
        "ch32v003f4u6",
        "memory-x",
        "embassy",
        "time-driver-tim2",
        "rt",
    ] }
embassy-executor = { version = "0.6.0", features = [
    "integrated-timers",
    "arch-spin",
    "executor-thread",
    "task-arena-size-128", # or better use nightly, but fails on recent Rust versions
] }
embassy-time = { version = "0.3.2" }

qingke = "0.5.0"
qingke-rt = "0.5.0"
```

For example, this will not build because `timeout` does not exist without  the `time` feature:
```rust
let i2c_config = i2c::Config::default();
println!("{}", i2c_config.timeout);
```

Thanks in advance!

**Comment by albertskog** [comment]
I noticed if I comment out all the `#[cfg(feature = "time")]` in i2c.rs, the file would not compile without adding `use embassy_time::{Duration, Instant};` so maybe this is still a work in progress? With that in place I got it running, but how is it intended to work? I might be able to contribute a PR if I can just understand what is missing!

**Reply by andelf** [reply]
My bad. A missing feature gate.

**Reply by albertskog** [reply]
Thank you, I'm glad I was not missing something obvious!
How should it be fixed, should `time` be added as a feature of `ch32-hal`?

**Reply by albertskog** [reply]
Something like this?
https://github.com/ch32-rs/ch32-hal/compare/main...albertskog:ch32-hal:main

**Reply by andelf** [reply]
It should be fixed now.

**Comment by albertskog** [comment]
Awesome, thank you!
And thanks for a great project!
